.github/workflows/mvn-verify-check.yml

```diff
@@ -140,4 +147,31 @@ jobs:
           -DskipTests
           -Dskip
           -Dmaven.javadoc.skip
-          
\ No newline at end of file
+
+  verify-modules-with-jdk17:
+    needs: get-shim-versions-from-dist
+    runs-on: ubuntu-latest
+    strategy:
+      matrix: ${{ fromJSON(needs.get-shim-versions-from-dist.outputs.sparkJDK17Versions) }}
+    steps:
+      - uses: actions/checkout@v3 # refs/pull/:prNumber/merge
+
+      - name: Setup Java and Maven Env
+        uses: actions/setup-java@v3
+        with:
+          distribution: adopt
+          java-version: 17
+
+      - name: Build JDK17
+        run: >
+          mvn -Dmaven.wagon.http.retryHandler.count=3 -B verify
+          -P 'individual,pre-merge'
+          -Dbuildver=${{ matrix.spark-version }}
+          -Dmaven.compiler.release=17
+          -Dmaven.compiler.source=17
+          -Dmaven.compiler.target=17
+          -Dscala.plugin.version=4.6.1
```

Comment:
Is there a reason we wouldn't just change the default for this in pom.xml rather than needing to override it here?  Does 4.6.1 cause problems when not on JDK17?

Reply:
I was following the doc, but yes let me try simply update the scala plugin version here

Reply:
tried to bump up version, but seems the newer one may not be fully compatible w/ jdk version below 9
```
Error:  -release is only supported on Java 9 and higher
Warning:  Unexpected javac output: javac: invalid flag: --release
Usage: javac <options> <source files>
use -help for a list of possible options.
Error:  NoPosition: javac: invalid flag: --release
Warning:  javac exited with exit code 2
...
Failed to execute goal net.alchim31.maven:scala-maven-plugin:4.6.1:compile (scala-compile-first) on project rapids-4-spark-sql_2.12: Execution scala-compile-first of goal net.alchim31.maven:scala-maven-plugin:4.6.1:compile failed: javac returned non-zero exit code: CompileFailed 
```

I will add the version update to jdk11+ profiles

Reply:
```
[INFO] Compiler bridge installed
[INFO] compiling 372 Scala sources and 59 Java sources to /home/runner/work/spark-rapids/spark-rapids/sql-plugin/target/spark321cdh/classes ...
Error:  /home/runner/work/spark-rapids/spark-rapids/sql-plugin/src/main/scala/com/nvidia/spark/rapids/MetaUtils.scala:112: value createUnintializedVector is not a member of com.google.flatbuffers.FlatBufferBuilder
Error:  /home/runner/work/spark-rapids/spark-rapids/sql-plugin/src/main/scala/com/nvidia/spark/rapids/MetaUtils.scala:236: value createUnintializedVector is not a member of com.google.flatbuffers.FlatBufferBuilder
Error:  /home/runner/work/spark-rapids/spark-rapids/sql-plugin/src/main/scala/com/nvidia/spark/rapids/MetaUtils.scala:253: type ByteBufferFactory is not a member of object com.google.flatbuffers.FlatBufferBuilder
Error:  /home/runner/work/spark-rapids/spark-rapids/sql-plugin/src/main/scala/com/nvidia/spark/rapids/MetaUtils.scala:304: value createUnintializedVector is not a member of com.google.flatbuffers.FlatBufferBuilder
Error:  /home/runner/work/spark-rapids/spark-rapids/sql-plugin/src/main/scala/com/nvidia/spark/rapids/MetaUtils.scala:320: overloaded method constructor FlatBufferBuilder with alternatives:
  (x$1: java.nio.ByteBuffer)com.google.flatbuffers.FlatBufferBuilder <and>
  ()com.google.flatbuffers.FlatBufferBuilder <and>
  (x$1: Int)com.google.flatbuffers.FlatBufferBuilder
 cannot be applied to (Int, com.nvidia.spark.rapids.DirectByteBufferFactory)
Error:  /home/runner/work/spark-rapids/spark-rapids/sql-plugin/src/main/scala/com/nvidia/spark/rapids/MetaUtils.scala:329: overloaded method constructor FlatBufferBuilder with alternatives:
  (x$1: java.nio.ByteBuffer)com.google.flatbuffers.FlatBufferBuilder <and>
  ()com.google.flatbuffers.FlatBufferBuilder <and>
  (x$1: Int)com.google.flatbuffers.FlatBufferBuilder
 cannot be applied to (Int, com.nvidia.spark.rapids.DirectByteBufferFactory)
Error:  /home/runner/work/spark-rapids/spark-rapids/sql-plugin/src/main/scala/com/nvidia/spark/rapids/MetaUtils.scala:343: overloaded method constructor FlatBufferBuilder with alternatives:
  (x$1: java.nio.ByteBuffer)com.google.flatbuffers.FlatBufferBuilder <and>
  ()com.google.flatbuffers.FlatBufferBuilder <and>
  (x$1: Int)com.google.flatbuffers.FlatBufferBuilder
 cannot be applied to (Int, com.nvidia.spark.rapids.DirectByteBufferFactory)
Error:  /home/runner/work/spark-rapids/spark-rapids/sql-plugin/src/main/scala/com/nvidia/spark/rapids/MetaUtils.scala:369: overloaded method constructor FlatBufferBuilder with alternatives:
  (x$1: java.nio.ByteBuffer)com.google.flatbuffers.FlatBufferBuilder <and>
  ()com.google.flatbuffers.FlatBufferBuilder <and>
  (x$1: Int)com.google.flatbuffers.FlatBufferBuilder
 cannot be applied to (Int, com.nvidia.spark.rapids.DirectByteBufferFactory)
Error: [ERROR] 8 errors found
```
we saw some compilation errors in cdh shims build after bump up scala.plugin.version to 4.6.1
let me revert scala.plugin.version to 4.3.0, and put 4.6.1 to jdk11+ profiles for now.

If needed, we can get back to fix compilation issues while using newer scala plugin later

Reply:
Scala plugin upgrade #5576 is also tracked as a task in #8181. And if the compilation issues affect just a few profiles, especially if just vendor profiles, we can pin the old version for them.

Reply:
> especially if just vendor profiles, we can pin the old version for them.

I would like not to apply the change in this PR. We can follow up in https://github.com/NVIDIA/spark-rapids/issues/5576